### ISSUE TITLE
About ablation experiments

### ISSUE THREAD

**Issue by HulkM** [yitongx/sinerf#4]
Hi, your work is very nice. It inspires me. However, I am confused about the result shown in Table 3, which indicates that either SIREN or MRS alone would negatively impact the results in the Fortress or Trex scenes. Surprisingly, when used together, they achieve the best result, surpassing even the baseline NeRF--. Can you please explain the details behind this? I would greatly appreciate it. Thanks.

**Comment by yitongx** [comment]
Hi, 

Thanks for your question!

On one hand, SIREN has a nature of producing smooth content. This may lead to blurriness on fine-detailed scenes, like Trex. This requires MRS to provide randomness and to distinguish different pixels.

On the other hand, MRS may not perform well alone on textureless scenes, like Fortress (the background is generally yellow color) since it may sample a homogenized ray batch that doesn't have adequate supervision, which may lead the optimization to fall into local minima. Original random sampling also has the same problem, but worse. SIREN can bring smoothness and help escape from local minima, helping MRS to release its capability.

In short conclusion: **it requires both SIREN and MRS to give the best performances** since each module is sensitive to scene content. We tested on other scenes, and the score differences are not as large as those on Fortress and Trex.

For more content, please refer to "Sec 4.4 Ablation Study" on Page 10  of [paper](https://arxiv.org/pdf/2210.04553.pdf) if you haven't read it. You can also find the explanation as to why the performance scores in Table 3 are ordered in that way.

Best.

**Comment by HulkM** [comment]
Thank you for your detailed and logical explanation. I now have a better understanding of it.

Regarding the results of your other ablation experiments in scenes with "mid-texture," how can they demonstrate that SIREN or MRS achieve better results than the baseline alone? In other words, SIREN and MRS can improve alone, but, together is the best?

Additionally, there is another work called GRAF that uses Gaussian-MLP. They mention that Sin-MLP also suffers from minima. Does this conflict with SiNeRF's conclusion?

Nevertheless, your work is very valuable, and I have gained a deeper understanding of the details. Thank you once again!

I am looking for the new work from your team!

**Comment by yitongx** [comment]
For "mid-texture" I assume you refer to scene content that is neither too fine-detailed nor too textureless. I remember using either SIREN or MRS independently can have slightly higher or lower performance scores than baseline on those scenes, just like what was shown in the Trex or Fortress ablation results. But the score differences were not that obvious and might flip with different initialization. So we just went through all 8 scenes in the LLFF dataset and selected the 2 most representative scenes for the analysis part of our paper. 

We believe this is the nature of the Joint Optimization within NeRFmm's framework, that the system is sensitive to scene content. Maybe a larger dataset than LLFF with more diverse scenes can be helpful on training and explaining such sensitivities. And we won't deny the existence of local minima in the SiNeRF, since we claim SiNeRF alleviates the local minima problem instead of completely solving it. 

GARF is a well-written paper that improves upon their previous work named BARF and other theoretical work. They tackle Joint Optimization within a different framework from NeRFmm. So yeah, SIREN-MLP does suffer from local minima and might not be good enough. It would be interesting to try Gaussian-MLP or other improvements on the encoding network. Unfortunately, we almost finished our work last year before the GARF paper was released. 

And I'm really glad that our work can give you some help or inspiration. Hope I have answered your doubts.

Best.

**Comment by HulkM** [comment]
Abeslutely nice answers!

Thanks for your time! and, thanks for SiNeRF again!

**Comment by yitongx** [comment]
With pleasure! Hope your research goes well. 

I will close this issue now. Feel free to reopen it when needed.

**Comment by yitongx** [comment]
With pleasure! Hope your research goes well.

I will close this issue now. Feel free to reopen it when needed.
